kubernetes/structure_cron_job.go

```diff
@@ -0,0 +1,111 @@
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"k8s.io/api/batch/v1beta1"
+)
+
+func flattenCronJobSpec(in v1beta1.CronJobSpec, d *schema.ResourceData) ([]interface{}, error) {
+	att := make(map[string]interface{})
+
+	att["concurrency_policy"] = in.ConcurrencyPolicy
+	if in.FailedJobsHistoryLimit != nil {
+		att["failed_jobs_history_limit"] = int(*in.FailedJobsHistoryLimit)
+	} else {
+		att["failed_jobs_history_limit"] = 1
+	}
+
+	att["schedule"] = in.Schedule
+
+	jobTemplate, err := flattenJobTemplate(in.JobTemplate, d)
+	if err != nil {
+		return nil, err
+	}
+	att["job_template"] = jobTemplate
+
+	if in.StartingDeadlineSeconds != nil {
+		att["starting_deadline_seconds"] = int64(*in.StartingDeadlineSeconds)
+	} else {
+		att["starting_deadline_seconds"] = 0
```

Comment:
The `else` block is not necessary. Zero is the default value for unset TypeInt. Also, there is a `Default` attribute on it's schema already.